tests/Kernel/Patterns/DateBlockPatternRenderingTest.php

```diff
@@ -0,0 +1,217 @@
+   */
+  public function testFromTimestamp(string $variant, int $date, array $assertions) {
+    $pattern = [
+      '#type' => 'pattern',
+      '#id' => 'date_block',
+      '#variant' => $variant,
+      '#fields' => [
+        'date' => DateValueObject::fromTimestamp($date),
+      ],
+    ];
+
+    $html = $this->renderRoot($pattern);
+    $this->assertRendering($html, $assertions);
+  }
+
+  /**
+   * Data provider for testFromArray.
+   *
+   * @return array
+   *   An array of test data arrays with assertions.
+   */
+  public function fromArrayDataProvider(): array {
+    return [
+      // Test "default" variant.
+      [
+        'variant' => 'default',
+        'date' => [
+          'day' => '24',
+          'month' => '09',
+          'year' => '1981',
+        ],
+        'assertions' => [
+          'count' => [
+            'div.ecl-date-block.ecl-date-block--default' => 1,
+            'div.ecl-date-block.ecl-date-block--cancelled' => 0,
+            'div.ecl-date-block.ecl-date-block--past' => 0,
+          ],
+          'equals' => [
+            'span.ecl-date-block__week-day' => 'Thursday',
+            'span.ecl-date-block__day' => '24',
+            'span.ecl-date-block__month' => '09',
+            'span.ecl-date-block__year' => '1981',
+          ],
+        ],
+      ],
+
+      // Test "cancelled" variant.
+      [
+        'variant' => 'cancelled',
+        'date' => [
+          'day' => '24',
+          'month' => '09',
+          'year' => '1981',
+        ],
+        'assertions' => [
+          'count' => [
+            'div.ecl-date-block.ecl-date-block--default' => 0,
+            'div.ecl-date-block.ecl-date-block--cancelled' => 1,
+            'div.ecl-date-block.ecl-date-block--past' => 0,
+          ],
+          'equals' => [
+            'span.ecl-date-block__week-day' => 'Thursday',
+            'span.ecl-date-block__day' => '24',
+            'span.ecl-date-block__month' => '09',
+            'span.ecl-date-block__year' => '1981',
+          ],
+        ],
+      ],
+
+      // Test "past" variant.
+      [
+        'variant' => 'past',
+        'date' => [
+          'day' => '24',
+          'month' => '09',
+          'year' => '1981',
+        ],
+        'assertions' => [
+          'count' => [
+            'div.ecl-date-block.ecl-date-block--default' => 0,
+            'div.ecl-date-block.ecl-date-block--cancelled' => 0,
+            'div.ecl-date-block.ecl-date-block--past' => 1,
+          ],
+          'equals' => [
+            'span.ecl-date-block__week-day' => 'Thursday',
+            'span.ecl-date-block__day' => '24',
+            'span.ecl-date-block__month' => '09',
+            'span.ecl-date-block__year' => '1981',
+          ],
+        ],
+      ],
+    ];
+  }
+
+  /**
+   * Data provider for testFromTimestamp.
+   *
+   * @return array
+   *   An array of test data arrays with assertions.
+   */
+  public function fromTimestampDataProvider(): array {
+    return [
```

Comment:
Why we are not using an external yml file, as `renderingDataProvider`?

Reply:
Same reason as previously explained.